plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHttpMetastoreFactory.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import com.google.inject.Inject;
+import io.airlift.units.Duration;
+import io.trino.filesystem.TrinoFileSystemFactory;
+import io.trino.plugin.hive.util.RetryDriver;
+import io.trino.spi.security.ConnectorIdentity;
+import org.weakref.jmx.Flatten;
+import org.weakref.jmx.Managed;
+
+import java.util.Optional;
+import java.util.concurrent.Executors;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+public class ThriftHttpMetastoreFactory
+        implements ThriftMetastoreFactory
+{
+    private final IdentityAwareMetastoreClientFactory metastoreClientFactory;
+    private TrinoFileSystemFactory fileSystemFactory;
+    private final ThriftMetastoreStats stats = new ThriftMetastoreStats();
+
+    @Inject
+    public ThriftHttpMetastoreFactory(
+            TrinoFileSystemFactory fileSystemFactory,
+            IdentityAwareMetastoreClientFactory metastoreClientFactory)
+    {
+        this.metastoreClientFactory = requireNonNull(metastoreClientFactory, "metastoreClientFactory is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+    }
+
+    @Override
+    public boolean isImpersonationEnabled()
+    {
+        return false;
+    }
+
+    @Override
+    public ThriftMetastore createMetastore(Optional<ConnectorIdentity> identity)
+    {
+        return new ThriftHiveMetastore(
+                identity,
+                fileSystemFactory,
+                metastoreClientFactory,
+                RetryDriver.DEFAULT_SCALE_FACTOR,
+                RetryDriver.DEFAULT_SLEEP_TIME,
+                RetryDriver.DEFAULT_SLEEP_TIME,
+                RetryDriver.DEFAULT_MAX_RETRY_TIME,
+                new Duration(10, MINUTES),
+                RetryDriver.DEFAULT_MAX_ATTEMPTS,
+                false,
+                false,
+                false,
+                false,
+                true,
+                stats,
+                Executors.newSingleThreadExecutor());
```

Comment:
Here you should probably bind the executor you bouded in ThriftMetastoreModule

Inject in the constructor

```
@ThriftHiveWriteStatisticsExecutor ExecutorService writeStatisticsExecutor
```

We should probably use `1` instead of `20` (what I originally suggested basd on the default value from `ThriftMetastoreConfig`) for the number of threads

Reply:
sorry, missed it. Done.